src/common/components/repositories-home/index.js

```diff
@@ -92,18 +96,18 @@ class RepositoriesHome extends Component {
   }
 
   render() {
-    // show spinner if snaps data was not yet fetched (snaps list is empty)
-    // (to avoid spinner during polling we are not checking `success` or `isFetching`
+    // show spinner if user has just signed in snaps data was not yet fetched
     //
-    // when snaps are loaded and user doesn't have any, they will be redirected
-    // to select repositories (so spinner won't be showing endlessly)
-    return !this.props.hasSnaps
+    // when snaps are loaded and user (who just signed in) doesn't have any,
+    // they will be redirected to select repositories
+    return (this.props.hasJustSignedIn && this.props.snaps.isFetching)
```

Comment:
I agree that this is sub-optimal and temporary workaround.

On the other hand it seems #780 is quite high in priority, so maybe we don't need to push this solution and just find proper solution to detect new users.